### PR TITLE
Deployment of hello app

### PR DESCRIPTION
**Pull request by SonOfLope**
### Description
Deployment of Hello app for the club App|ets : https://github.com/ApplETS/Backend-Hello/tree/main

We are currently deploying the backend of the app with a postgresql database and a redis server.
### Checklist

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

- [ ] Requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/applets/hello/base/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hello
+  namespace: applets-hello
```
Comment (Simon-Boyer) [inline apps/applets/hello/base/deployment.yaml:5]:
```suggestion
```

File: apps/applets/hello/base/redis/service.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis
+  namespace: applets-hello
```
Comment (Simon-Boyer) [inline apps/applets/hello/base/redis/service.yaml:5]:
```suggestion
```

File: apps/applets/hello/base/redis/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: redis
+  namespace: applets-hello
```
Comment (Simon-Boyer) [inline apps/applets/hello/base/redis/statefulset.yaml:5]:
```suggestion
```

File: apps/applets/hello/base/service.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: hello-svc
+  namespace: applets-hello
```
Comment (Simon-Boyer) [inline apps/applets/hello/base/service.yaml:5]:
```suggestion
```

File: apps/applets/hello/prod/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
```
Comment (Simon-Boyer) [inline apps/applets/hello/prod/kustomization.yaml:3]:
```suggestion

namespace: applets-hello

bases:
  - ../base
  
```

File: apps/applets/hello/base/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    spec:
+      serviceAccountName: secrets-reader
+      containers:
+        - name: hello
+          image: ghcr.io/applets/backend-hello:main
+          imagePullPolicy: Always
+          command: ['/bin/sh', '-c', 'source /vault/secrets/hello-env && dotnet api.core.dll']
+          ports:
+            - containerPort: 8080
+          volumeMounts:
```
Comment (Simon-Boyer) [inline apps/applets/hello/base/deployment.yaml:37]: Adapter selon la prevision de consommation de ressources et de droits nécessaires

```suggestion
          resources:
            requests:
              cpu: 100m
              memory: 128Mi
            limits:
              cpu: 500m
              memory: 256Mi
              ephemeral-storage: 1Gi
          securityContext:
            allowPrivilegeEscalation: false
            readOnlyRootFilesystem: false
            runAsNonRoot: true
            runAsUser: 10001
            runAsGroup: 10001
            capabilities:
              drop:
                - ALL
            seccompProfile:
              type: RuntimeDefault
          volumeMounts:
```

**Review by Simon-Boyer — CHANGES_REQUESTED** [review]
J'ai mis un changement dans le deploiement de l'app avec le security context et les limites de ressources, faudrait faire la meme chose pour le statefulset postgres et redis

**Comment by SonOfLope** [comment]
> J'ai mis un changement dans le deploiement de l'app avec le security context et les limites de ressources, faudrait faire la meme chose pour le statefulset postgres et redis

@Simon-Boyer all done!

**Review by Simon-Boyer — CHANGES_REQUESTED** [review]
Kustomize fails with this:

> error: accumulating resources: accumulation err='accumulating resources from '../base/': '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base' must resolve to a file': recursed accumulation of path '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base': accumulating resources: accumulation err='accumulating resources from 'postgres/pv.yaml': evalsymlink failure on '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml' : lstat /home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml' : lstat /home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml: no such file or directory

Looks like `base/postgres/pv.yaml` is missing

**Comment by SonOfLope** [comment]
> Kustomize fails with this:
> 
> > error: accumulating resources: accumulation err='accumulating resources from '../base/': '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base' must resolve to a file': recursed accumulation of path '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base': accumulating resources: accumulation err='accumulating resources from 'postgres/pv.yaml': evalsymlink failure on '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml' : lstat /home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml' : lstat /home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml: no such file or directory
> 
> Looks like `base/postgres/pv.yaml` is missing

oops yeah je l'ai enlever pcq avec le statefulset pv template il était pas nécessaire.

**Comment by SonOfLope** [comment]
> Kustomize fails with this:
> 
> > error: accumulating resources: accumulation err='accumulating resources from '../base/': '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base' must resolve to a file': recursed accumulation of path '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base': accumulating resources: accumulation err='accumulating resources from 'postgres/pv.yaml': evalsymlink failure on '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml' : lstat /home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml' : lstat /home/simon/Repos/Plateforme-Cedille/apps/applets/hello/base/postgres/pv.yaml: no such file or directory
> 
> Looks like `base/postgres/pv.yaml` is missing

done

**Review by Simon-Boyer — APPROVED** [review]

**Pull request merged**